tests/extmod_hardware/machine_uart_irq_break.py

```diff
@@ -0,0 +1,67 @@
+# Test machine.UART.IRQ_BREAK firing after a break is received.
+#
+# IMPORTANT: This test requires hardware connections: the UART TX and RX
+# pins must be wired together.
+
+try:
+    from machine import UART
+
+    UART.IRQ_BREAK
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+# Configure pins based on the target.
+if "esp32" in sys.platform and not (
+    "ESP32S2" in sys.implementation._machine or "ESP32C3" in sys.implementation._machine
+):
+    # The ESP32 needs separate UART instances for the test
+    recv_uart_id = 1
+    recv_tx_pin = 14
+    recv_rx_pin = 5
+    send_uart_id = 2
+    send_tx_pin = 4
+    send_rx_pin = 12
+elif "rp2" in sys.platform:
+    recv_uart_id = 0
+    send_uart_id = 0
+    recv_tx_pin = "GPIO0"
+    recv_rx_pin = "GPIO1"
+else:
+    print("SKIP")
+    raise SystemExit
+
+
+def irq(u):
+    print("IRQ_BREAK:", bool(u.irq().flags() & u.IRQ_BREAK), "data:", u.read(1))
+
+
+# Test that the IRQ is called for each break received.
+for bits_per_s in (2400, 9600, 57600):
+    if recv_tx_pin is None:
```

Comment:
This block of if-code is not needed.

Reply:
I kept it for a possible future board that does not allow specifying the rx/tx pins. I still wonder why the MIMXRT lib does not provide the `break` event.

Reply:
I would rather go the other way and make sure all ports allow specifying the tx/rx pins (not required but allowed).  That makes ports more consistent.

For a port that can't change the tx/rx pin then all it needs to do is verify that the passed-in pin matches the pre-configured pin.

Anyway, all that is for another day.  For now I don't think we need this bit of code in the test.

Reply:
Removed